client/js/components/SmallUnsubscribeButton/SmallUnsubscribeButton.js

```diff
@@ -1,12 +1,54 @@
 import React, {Component} from 'react';
 import styles from './styles';
-import {TouchableOpacity, Text} from 'react-native';
+import {TouchableOpacity, Text, View} from 'react-native';
+import {Mutation} from '@apollo/react-components';
+import client from '../../config/api';
+import gql from 'graphql-tag';
 
-const SmallUnsubscribeButton = props => {
+const CAMPAIGN_UNSUBSCRIBE = gql`
+  mutation updateUser($campaignid: ID!, $userid: ID!) {
+    updateUser(
+      data: {campaigns: {disconnect: {id: $campaignid}}}
+      where: {id: $userid}
+    ) {
+      id
+      name
+      campaigns {
+        id
+        title
+      }
+    }
+  }
+`;
+
+const SmallUnsubscribeButton = ({CampaignId, userid}) => {
   return (
-    <TouchableOpacity style={styles.button}>
-      <Text style={styles.buttonText}>Unsubscribe</Text>
-    </TouchableOpacity>
+    <Mutation mutation={CAMPAIGN_UNSUBSCRIBE} client={client}>
+      {updateUser => {
+        return (
+          <View
+            style={{
+              height: '70%',
+              width: '24%',
+            }}>
+            <TouchableOpacity
+              onPress={async () => {
+                try {
+                  const editSubscribe = await updateUser({
+                    variables: {
+                      campaignid: CampaignId,
+                      userid: userid,
+                    },
+                  });
+                } catch (e) {}
```

Comment:
what should we do when there is an error?

Reply:
Done